app/code/Magento/CatalogGraphQl/Model/Resolver/Product/SpecialPrice.php

```diff
@@ -28,7 +28,10 @@ public function resolve(Field $field, $context, ResolveInfo $info, array $value
         /** @var PricingSpecialPrice $specialPrice */
         $specialPrice = $product->getPriceInfo()->getPrice(PricingSpecialPrice::PRICE_CODE);
 
-        if ($specialPrice->getValue()) {
+        if ((!$product->hasData('can_show_price')
+                || ($product->hasData('can_show_price') && $product->getData('can_show_price') === true)
```

Comment:
there's no constant in core for this right? if not..leave it like this

Reply:
No, no constant as far I can see.